ql/src/test/org/apache/hadoop/hive/metastore/txn/TestTxnHandler.java

```diff
@@ -1544,21 +1546,68 @@ public void testRetryableRegex() throws Exception {
     Assert.assertTrue("regex should be retryable", result);
   }
 
-  @Test
-  public void testReplOpenTxn() throws Exception {
-    int numTxn = 50000;
+  private List<Long> openTxnForTest(int startId, int numTxn, String replPolicy) throws MetaException, NoSuchTxnException {
     conf.setIntVar(HiveConf.ConfVars.HIVE_TXN_MAX_OPEN_BATCH, numTxn);
+    int lastId = startId + numTxn;
     OpenTxnRequest rqst = new OpenTxnRequest(numTxn, "me", "localhost");
-    rqst.setReplPolicy("default.*");
-    rqst.setReplSrcTxnIds(LongStream.rangeClosed(1, numTxn)
+    rqst.setReplPolicy(replPolicy);
+    rqst.setReplSrcTxnIds(LongStream.rangeClosed(startId, lastId)
             .boxed().collect(Collectors.toList()));
     OpenTxnsResponse openedTxns = txnHandler.openTxns(rqst);
     List<Long> txnList = openedTxns.getTxn_ids();
     assertEquals(txnList.size(), numTxn);
     for (long i = 0; i < numTxn; i++) {
       long txnId = txnList.get((int) i);
-      assertEquals(i+1, txnId);
+      assertEquals(i + startId, txnId);
+    }
+    return txnList;
+  }
+
+  @Test
+  public void testReplOpenTxn() throws Exception {
+    int numTxn = 50000;
+    List<Long> txnList = openTxnForTest(1, numTxn, "default.*");
+    assert(txnList.size() == numTxn);
+    txnHandler.abortTxns(new AbortTxnsRequest(txnList));
```

Comment:
What are we testing here? No verification after abort.

Reply:
the assert is added in the common function oopenTxnFortest